src/MPAGSCipher/VigenereCipher.cpp

```diff
@@ -0,0 +1,72 @@
+#include "VigenereCipher.hpp"
+#include "Alphabet.hpp"
+#include "CipherMode.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+VigenereCipher::VigenereCipher ( const std::string& key )
+{
+    // Set the given key
+    setKey(key);
+}
+
+void VigenereCipher::setKey(const std::string& key) {
+
+    // Store the original key
+    key_ = key;
+
+    // Make sure the key is uppercase
+    std::transform(std::begin(key_), std::end(key_), std::begin(key_), ::toupper);
+
+    // Remove non-alphabet characters
+    key_.erase(std::remove_if(std::begin(key_), std::end(key_), [](char c) { return !std::isalpha(c); }), std::end(key_));
+
+    // Check if the key is empty and replace with default if so
+    if (key_.empty()){
+        key_ = "MYKEY";
+    }
+    // loop over the key
+    for (char c : key_){
+
+        // Find the letter position in the alphabet
+        const size_t position = Alphabet::alphabet.find(c);
+
+        // Create a CaesarCipher using this position as a key
+        CaesarCipher caesarCipher(position);
+
+        // Insert a std::pair of the letter and CaesarCipher into the lookup
+        charLookup_.insert(std::make_pair(c, caesarCipher));
+    }
+}
+
+std::string VigenereCipher::applyCipher(const std::string& inputText,
+                                        const CipherMode cipherMode) const
+{
+    std::string outputText = "";
```

Comment:
Suggest to reserve space for the output:
```suggestion
    std::string outputText = "";
    outputText.reserve( inputText.size() );
```